src/core_functions/scalar/math/numeric.cpp

```diff
@@ -1323,4 +1332,206 @@ ScalarFunctionSet LeastCommonMultipleFun::GetFunctions() {
+	return UINT32_C(0xFFFFFFFF) / rhs + 1;
+}
+
+static uint16_t ComputeM(uint8_t rhs) {
+	return UINT32_C(0xFFFF) / rhs + 1;
+}
+
+static uint64_t ComputeM(int32_t rhs) {
+	if (rhs < 0) {
+		rhs = -rhs;
+	}
+	return UINT64_C(0xFFFFFFFFFFFFFFFF) / rhs + 1 + ((rhs & (rhs - 1)) == 0 ? 1 : 0);
+}
+
+static uint32_t ComputeM(int16_t rhs) {
+	if (rhs < 0) {
+		rhs = -rhs;
+	}
+	return UINT32_C(0xFFFFFFFF) / rhs + 1 + ((rhs & (rhs - 1)) == 0 ? 1 : 0);
+}
+
+static uint16_t ComputeM(int8_t rhs) {
+	if (rhs < 0) {
+		rhs = -rhs;
+	}
+	return UINT32_C(0xFFFF) / rhs + 1 + ((rhs & (rhs - 1)) == 0 ? 1 : 0);
+}
+
+// fastdiv computes (a / d) given precomputed M for d>1
+inline static uint64_t Fastdiv(uint64_t a, uhugeint_t m, uint64_t d) {
+	// Compute the bottom 32 bits, then the top 32 bits, add them, and only keep the top 32 bits
+	// The reason we need the bottom 32 bits, is because the carry will roll into the top 32 bits.
+	return ((((m & 0xFFFFFFFFFFFFFFFF) * a) >> 64) + ((m >> 64) * a)).upper;
+}
+
+// fastdiv computes (a / d) given precomputed M for d>1
+inline static uint32_t Fastdiv(uint32_t a, uint64_t m, uint32_t d) {
+	// Compute the bottom 32 bits, then the top 32 bits, add them, and only keep the top 32 bits
+	// The reason we need the bottom 32 bits, is because the carry will roll into the top 32 bits.
+	return ((((m & 0xFFFFFFFF) * a) >> 32) + ((m >> 32) * a)) >> 32;
+}
+
+inline static uint16_t Fastdiv(uint16_t a, uint32_t m, uint16_t d) {
+	// Compute the bottom 32 bits, then the top 32 bits, add them, and only keep the top 32 bits
+	// The reason we need the bottom 32 bits, is because the carry will roll into the top 32 bits.
+	return ((((m & 0xFFFF) * a) >> 16) + ((m >> 16) * a)) >> 16;
+}
+
+inline static uint8_t Fastdiv(uint8_t a, uint16_t m, uint8_t d) {
+	// Compute the bottom 32 bits, then the top 32 bits, add them, and only keep the top 32 bits
+	// The reason we need the bottom 32 bits, is because the carry will roll into the top 32 bits.
+	return ((((m & 0xFF) * a) >> 8) + ((m >> 8) * a)) >> 8;
+}
+
+static int32_t Fastdiv(int32_t a, uint64_t m, int32_t d) {
+	hugeint_t lhs;
+	if (a < 0) {
+		lhs = hugeint_t(0xffffffffffffffff, a);
+	} else {
+		lhs = hugeint_t(0, a);
+	}
+	uint64_t highbits = Hugeint::Multiply<false>(lhs, hugeint_t(0, m)).upper;
+	highbits += (a < 0 ? 1 : 0);
+	if (d < 0) {
+		return -(int32_t)(highbits);
+	}
+	return (int32_t)(highbits);
+}
+
+static int16_t Fastdiv(int16_t a, uint32_t m, int16_t d) {
+	int64_t lhs = a;
+	if (a < 0) {
+		lhs |= int64_t(0xffffffff) << 32;
+	}
+	uint32_t highbits = (lhs * m) >> 32;
+	highbits += (a < 0 ? 1 : 0);
+	if (d < 0) {
+		return -(int16_t)(highbits);
+	}
+	return (int16_t)(highbits);
+}
+
+static int8_t Fastdiv(int8_t a, uint16_t m, int8_t d) {
+	int32_t lhs = a;
+	if (a < 0) {
+		lhs |= int32_t(0xffffffff) << 16;
+	}
+	uint16_t highbits = (lhs * m) >> 16;
+	highbits += (a < 0 ? 1 : 0);
+	if (d < 0) {
+		return -(int8_t)(highbits);
+	}
+	return (int8_t)(highbits);
```

Comment:
Could you replace the casting here with `UnsafeNumericCast`? This is a helper function that does an efficient static cast with our release builds, but performs over-/underflow checking for debug builds, which helps with finding bugs.